dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java

```diff
@@ -77,5 +82,43 @@ public final Template getTemplate(IHTMLPage page, PageMode mode) {
                 + page.getLanguageId() + "." + VelocityType.HTMLPAGE.fileExtension);
     }
 
+    protected void handleParseError (final ParseErrorException e, final String name, final User user) {
 
+        try {
+
+            final SystemMessageBuilder systemMessageBuilder = new SystemMessageBuilder();
+            final StringBuilder message                     = new StringBuilder();
+            final String errorMessage                       = WordUtils.wrap
+                    (e.getMessage(), 15, "<br/>", false);
+
+            message.append("<h3>").append("Parsing Error").append("</h3>")
+                    .append("<b>Name: </b>").append("<br/>&nbsp;&nbsp;")
+                    .append(name).append("<br/>")
+                    .append("<b>Template: </b>").append("<br/>&nbsp;&nbsp;")
+                    .append(e.getTemplateName()).append("<br/>")
+
+                    .append("<b>Invalid Syntax: </b>").append("<br/>&nbsp;&nbsp;")
+                    .append(e.getInvalidSyntax()).append("<br/>")
+
+                    .append("<b>Column Number: </b>").append("<br/>&nbsp;&nbsp;")
+                    .append(e.getColumnNumber()).append("<br/>")
+
+                    .append("<b>Line Number: </b>").append("<br/>&nbsp;&nbsp;")
+                    .append(e.getLineNumber()).append("<br/>")
+
+                    .append("<pre>")
+                    .append(errorMessage)
+                    .append("</pre>");
+
+            systemMessageBuilder.setMessage(message.toString())
+                    .setLife(DateUtil.FIVE_SECOND_MILLIS)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setSeverity(MessageSeverity.ERROR);
+
+            SystemMessageEventUtil.getInstance().
+                    pushMessage(systemMessageBuilder.create(), Arrays.asList(user.getUserId()));
+        } catch (Exception ex) {
+            Logger.error(this, ex.getMessage(), ex);
```

Comment:
We should include the "name" in the error message, ideally the Identifier as well. Useful for Support.

Reply:
ok